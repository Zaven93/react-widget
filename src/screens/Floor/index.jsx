import { useContext } from "react";

import { NavContext } from "../../context/navContext";
import { BuildingContext } from "../../context/BuildingContext";
import { floorTestId } from "../../tests/constants";
import FilterContainer from "../../components/FilterContainer";
import FilterUnitType from "../../components/FilterUnitType";
import FilterFloor from "../../components/FilterFloor";

const Floor = ({ hash, nextLink, prevLink }) => {
  const navContext = useContext(NavContext);
  const { reset, buildingState } = useContext(BuildingContext);

  console.log("Building state is", buildingState);

  return (
    <div
      className="building-widget__container relative"
      data-testid={floorTestId}
    >

      <div className="lg:hidden">
        <div className="building-widget__logo text-center py-4 mx-auto mb-3">
          <img
            src="https://res.cloudinary.com/dpwbaabx2/image/upload/v1637841004/euz7n8h5jsg3cd8nh8jc.png"
            alt="description"
            loading="lazy"
            width="193"
            height="86"
          />
        </div>

        <div className="building-widget-block">
          <button
            className="building-widget__btn-back inline-block relative"
            disabled={!prevLink}
            onClick={() => navContext.triggerTransition(hash, prevLink)}
          >
            Back
          </button>
        </div>
      </div>

      <div className="lg:flex lg:flex-row-reverse">
        <div className="lg:w-8/12 xl:w-9/12 lg:pl-2">
          <div className="building-widget__img-block">
            <img
              src={buildingState.chosenFloorImage}
              alt="description"
              loading="lazy"
              width="1436"
              height="1080"
            />
          </div>
        </div>

        <div className="lg:w-4/12 xl:w-3/12 lg:pr-2">
          <div className="building-widget-block mx-auto">
            <div className="hidden lg:block">
              <div className="building-widget__logo text-center pb-5 mx-auto mb-16">
                <img
                  src="https://res.cloudinary.com/dpwbaabx2/image/upload/v1637841004/euz7n8h5jsg3cd8nh8jc.png"
                  alt="description"
                  loading="lazy"
                  width="193"
                  height="86"
                />
              </div>
            </div>

            <FilterContainer
              withBack={true}
              prevLink={prevLink}
              hash={hash}
              title={`${buildingState.chosenFloor}st Floor`}
            >
              <FilterFloor title="Select a floor" />
              <FilterUnitType reset={reset} title="Filter by unit type" />
            </FilterContainer>
          </div>
        </div>
      </div>

      <div className="navigation-button-container">
        <button
          disabled={!prevLink}
          onClick={() => navContext.triggerTransition(hash, prevLink)}
        >
          Prev
        </button>
        <button
          disabled={!nextLink}
          onClick={() => navContext.triggerTransition(hash, nextLink)}
        >
          Next
        </button>
      </div>
    </div>
  );
};

export default Floor;
