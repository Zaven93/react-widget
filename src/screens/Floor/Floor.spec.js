import { render, screen } from "@testing-library/react";

import Floor from ".";

describe("Floor component", () => {
  it("Should have navigation button and text", () => {
    const { getByText, container } = render(
      <Floor nextLink={true} prevLink={true} />
    );

    const nextButton = getByText("Next");
    const prevButton = getByText("Prev");

    expect(nextButton).toBeTruthy();
    expect(nextButton.closest("button")).toBeEnabled();

    expect(prevButton).toBeTruthy();
    expect(prevButton.closest("button")).toBeEnabled();

    expect(container).toMatchSnapshot();
  });
});
