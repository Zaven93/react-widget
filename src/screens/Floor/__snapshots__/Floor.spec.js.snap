// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Floor component Should have navigation button and text 1`] = `
<div>
  <div
    data-testid="/floor-screen/"
  >
    <h1>
      Floor
    </h1>
    <div
      class="button-container"
    >
      <button>
        Prev
      </button>
      <button>
        Next
      </button>
    </div>
  </div>
</div>
`;
